src/api/invitation.ts

```diff
@@ -420,6 +421,14 @@ export class Invitation extends Session {
   public _onCancel(message: IncomingRequestMessage): void {
     this.logger.log("Invitation._onCancel");
 
+    // validate state for CANCEL delegation
+    if (this.state === SessionState.Initial || this.state === SessionState.Establishing) {
```

Comment:
Why do we need to check the State not just send the Cancel?
What problem it solves?

Reply:
After talking with Eric from SIP.js he said he'll only accept this addition only if I won't try to bypass the RFC.
As CANCEL should not arrive in ESTABLISHED state, and if it does according the RFC it should ignore it and not process it at all. 
That's why the delegation part will only work in INITIAL and ESTABLISHING.

Initial - is when the CT is still ringing.
Establishing - is when the CT SIP.js is still structing the 200OK but hasn't sent it out yet.
Established - is when the 200OK is already sent.